src/installation-info/ck/getCKPremiumBundleInstallationInfo.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import type { BundleInstallationInfo } from '../types';
+
+import { isCSSClassDefined } from '@/utils/isCSSClassDefined';
+import { isSemanticVersion } from '@/utils/isSemanticVersion';
+
+/**
+ * Returns information about the premium CKEditor bundle installation.
+ */
+export function getCKPremiumBundleInstallationInfo(): BundleInstallationInfo | null {
+	const { CKEDITOR_VERSION, CKEDITOR_PREMIUM_FEATURES } = window;
+
+	if ( !isSemanticVersion( CKEDITOR_VERSION ) ) {
+		return null;
+	}
+
+	// Global `CKEDITOR_PREMIUM_FEATURES` is set only in CDN builds.
+	if ( CKEDITOR_PREMIUM_FEATURES || document.querySelector( 'link[href*="ckeditor5-premium-features.css"]' ) ) {
+		return {
+			source: 'cdn',
+			version: CKEDITOR_VERSION
+		};
+	}
+
+	// It's not injected from the CDN. Let's check availability of the stylesheets.
+	// If there is class defined in the stylesheet, it means that the stylesheet is loaded
+	// and if bundle is not loaded from the CDN, it's loaded from NPM.
+	if ( isCSSClassDefined( 'ck-suggestion' ) ) {
+		return {
+			source: 'npm',
+			version: CKEDITOR_VERSION
+		};
+	}
```

Comment:
I'd prefer not to have this detection at all than to have it rely on a presence or absence of a CSS class, which could change at any time.

Also, why wouldn't `ck-suggestion` not be present in the [CDN version](https://cdn.ckeditor.com/ckeditor5-premium-features/43.0.0/ckeditor5-premium-features.css)?

Reply:
> I'd prefer not to have this detection at all than to have it rely on a presence or absence of a CSS class, which could change at any time.

I'd prefer too, but is there a better way to detect it? 

> Also, why wouldn't ck-suggestion not be present in the [CDN version](https://cdn.ckeditor.com/ckeditor5-premium-features/43.0.0/ckeditor5-premium-features.css)?

It's, but this condition will never be reached in CDN mode because ` CKEDITOR_PREMIUM_FEATURES ` if checked in previous step

Reply:
This check is no longer present